### PR TITLE
iox-#278 Extend Logger class with IsEnabled() and  Introduce template…

### PR DESCRIPTION
**Pull request by dkroenke**
Code contributed @prasannabhat and @nihalchari

### CODE REVIEW

File: iceoryx_utils/include/iceoryx_utils/log/logger.inl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_LOG_LOGGER_INL
+#define IOX_UTILS_LOG_LOGGER_INL
+namespace iox
+{
+namespace log
+{
+inline bool Logger::IsEnabled(const LogLevel logLevel) const noexcept
```
Comment (elBoberido) [inline iceoryx_utils/include/iceoryx_utils/log/logger.inl:21]: the `logger.inl` is not necessary. Please move this function to `logger.cpp`

**Review by elBoberido — CHANGES_REQUESTED** [review]

File: iceoryx_utils/include/iceoryx_utils/log/logger.inl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    if (logLevel <= m_logLevel.load(std::memory_order_relaxed))
+    {
+        return true;
+    }
+    return false;
+}
+
+} // namespace log
+} // namespace iox
+#endif // IOX_UTILS_LOG_LOGGER_INL
```
Comment (marthtz) [inline iceoryx_utils/include/iceoryx_utils/log/logger.inl:32]: Missing newline at end of file.

**Review by marthtz — CHANGES_REQUESTED** [review]

File: iceoryx_utils/include/iceoryx_utils/log/logger.inl (hunk 1 above)
Comment (nihalchari) [reply iceoryx_utils/include/iceoryx_utils/log/logger.inl]: moved

File: iceoryx_utils/include/iceoryx_utils/log/logger.inl (hunk 2 above)
Comment (nihalchari) [reply iceoryx_utils/include/iceoryx_utils/log/logger.inl]: this file is removed

**Review by elBoberido — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]

**Pull request merged**